nixos/modules/services/hardware/lirc.nix

```diff
@@ -0,0 +1,92 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.lirc;
+in {
+
+  ###### interface
+
+  options = {
+    services.lirc = {
+
+      enable = mkEnableOption "LIRC daemon";
+
+      options = mkOption {
+        type = types.lines;
+        example = ''
+          [lircd]
+          nodaemon = False
+        '';
+        description = "LIRC default options (lirc_options.conf)";
+      };
+
+      config = mkOption {
+        type = types.lines;
+        description = "Configuration for lircd to load, see man:lircd.conf(5) for details";
+      };
+
+      extraArguments = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = "Extra arguments to lircd.";
+      };
+
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    # Note: LIRC executables raises a warning, if lirc_options.conf do not exists
+    environment.etc."lirc/lirc_options.conf".text = cfg.options;
+
+    environment.systemPackages = [ pkgs.lirc ];
+
+    system.activationScripts.lirc = ''
+      if ! [ -d /run/lirc ]; then
+        umask 027
+        mkdir -p /run/lirc
+        chown lirc:lirc /run/lirc
+      fi
+    '';
```

Comment:
There are a few things here:

1. the mkdir/chown lines can be combined using ```install``` instead, *but*
2. it's superfluous as the /run/lirc directory will be automatically created by the ListenStream bit on your socket configuration ref man systemd.socket.

Reply:
The folder is created to grant lircd write access. It creates a pid file and removes for any reason the socket.
I tried to follow your hints:
1. I replace it with "install -o vdr -g vdr -m 0750 -d /run/lirc", but for any reason, the folder is owned by root?!
2. If systemd creates the folder, the owner is root. So I used ExecStartPost to change the owner, which is to late for any reason?!

Thanks for your review, but I do not find an alternative solution.

Reply:
Since ```lirc.socket``` creates the directory, my guess is that it overrides what you are doing in the .service.

You probably need to set the following in the .socket definition instead:
```
DirectoryMode = "750";
SocketUser = "lirc";
SocketGroup = "lirc";
```

Then you can get rid of the script.

Reply:
```man systemd.socket``` is your friend.

Reply:
Thanks for your suggestions, but
 - SocketUser is already set,
 - SocketGroup is not needed, because Systemd takes the primary group of user given at SocketUser
 - DirectoryMode is by default 755, which is good enough.

Lircd needs to be owner (or at least "create rights") on the folder /run/lirc.